File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (head == nullptr) {
            return head;
        }
        
        ListNode* curNode = head->next;
        ListNode* prevNode = head;

        while (curNode) {
            if (curNode->val == prevNode->val) {
                prevNode->next = curNode->next;
                delete curNode;
                curNode = prevNode->next;
            }
            else {
                prevNode = curNode;
                curNode = curNode->next;
            }
        }

        return head;
    }
};

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        
        curPointer = head.next
        prevPointer = head

        while curPointer:
            if curPointer.val == prevPointer.val:
                prevPointer.next = curPointer.next
                curPointer = prevPointer.next
            else:
                prevPointer = curPointer
                curPointer = curPointer.next
        
        return head
        